src/routes/webhooks.ts

```diff
@@ -109,4 +141,41 @@ export default async function (fastify: FastifyInstance): Promise<void> {
       return res.status(204).send();
     },
   });
+
+  fastify.post<{
+    Body: {
+      userId: string;
+      marketingCtaId: string;
+    };
+  }>('/customerio/marketing_cta', {
+    config: {
+      rawBody: true,
+    },
+    handler: async (req, res) => {
+      const valid = verifyCIOSignature(process.env.CIO_WEBHOOK_SECRET, req);
+      if (!valid) {
+        return res.status(403).send({ error: 'Invalid signature' });
+      }
+
+      try {
+        const { userId, marketingCtaId } = req.body;
+
+        const con = await createOrGetConnection();
+        await con.getRepository(UserMarketingCta).upsert(
+          {
+            userId: userId,
+            marketingCtaId: marketingCtaId,
+          },
+          ['userId', 'marketingCtaId'],
+        );
+
+        await getMarketingCta(con, logger, userId);
+
+        return res.send({ success: true });
+      } catch (error) {
+        logger.error(error, 'Error processing CIO webhook');
```

Comment:
We usually send `logger.error({ err: error }, 'Error processing CIO webhook'');`, I think because GC then extracts from `err`?